apps/src/blockly/googleBlocklyWrapper.js

```diff
@@ -516,6 +518,19 @@ function initializeBlocklyWrapper(blocklyInstance) {
     return this.JavaScript;
   };
 
+  blocklyWrapper.addNonRunnableWorkspace = function (workspace) {
+    this.nonRunnableWorkspaces.push(workspace.id);
+  };
+
+  // By default, a workspace is runnable unless we have specifically marked it
+  // as not runnable when creating a read only block space.
+  blocklyWrapper.isRunnableWorkspace = function (workspace) {
+    if (this.nonRunnableWorkspaces.includes(workspace.id)) {
```

Comment:
nit: I think we could just return `this.nonRunnableWorkspaces.includes(workspace.id)`.

Reply:
I guess we'd have to return `!this.nonRunnableWorkspaces.includes(id)`, which I guess is hard to read, so nevermind 😅

Reply:
I'm updating this to be `isEmbeddedWorkspace` so I can do a version of your first suggestion now!